### ISSUE TITLE
Review usage of `aria-label`

### ISSUE THREAD

**Issue by mkrause** [fortanix/baklava#379]
We're currently using `aria-label` in places where we should instead just rely on the text content. Need to review these and see if we can remove them or refactor them so the text content works better for assistive technology users.

References:
- https://www.smashingmagazine.com/2025/06/what-i-wish-someone-told-me-aria/#aria-label-is-tricky

**Comment by mkrause** [comment]
Thanks @nighto.

I think this is missing a few things, in particular the usages of `aria-label` where the there is both an `aria-label` as well as content inside the element, causing it to be pronounced twice. For example:

<img width="731" height="328" alt="Image" src="https://github.com/user-attachments/assets/3331bde6-8490-45e4-9b39-6fbc2678b902" />

The `aria-label` and the content of the button will both be pronounced. If we want to prevent this, we'd either have to (1) remove the `aria-label` and make the child content be what should be pronounced (this would be ideal, though not always feasible), or (2) keep the `aria-label` but make the element content hidden to assistive technology through something like:

```
<button aria-label="My label">
  <span aria-hidden>My visual label</span>
</button>
```

**Comment by nighto** [comment]
Yeah, that makes sense. Will fix those cases with `aria-hidden` inside.
